src/http/__tests__/users.test.ts

```diff
@@ -46,6 +46,64 @@ describe("POST /users", () => {
     });
     expect(user).toBeDefined();
   });
+
+  it('should return 400 for invalid username', async () => {
+    const invalidUser = {
+      username: 'a', // muito curto
+      email: 'test@example.com',
+      birthDate: '2000-01-01'
+    };
+
+    const response = await fastify.inject({
+      method: 'POST',
+      url: '/users',
+      payload: invalidUser
+    });
+
+    expect(response.statusCode).toBe(400);
+    const error = JSON.parse(response.payload);
+    expect(error.message).toBe('Validation failed');
+    expect(error.details.username._errors).toContain('Username must be at least 3 characters');
```

Comment:
é importante testar o restante das mensagens de erro, como `"Username must contain only letters, numbers, underscores, and periods"`. não há problema em adicioná-los nesse mesmo caso de teste, mas muitas vezes será interessante testá-los em separado,